schema/bom-1.6.proto

```diff
@@ -0,0 +1,1956 @@
+      repeated string openCre = 6;
+      // The optional `bom-ref` to a parent requirement. This establishes a hierarchy of requirements. Top-level requirements must not define a parent. Only child requirements should define parents.
+      optional string parent = 7;
+      // Specifies optional, custom, properties
+      repeated Property properties = 8;
+      // External references provide a way to document systems, sites, and information that may be relevant, but are not included with the BOM. They may also establish specific relationships within or external to the BOM.
+      repeated ExternalReference externalReferences = 9;
+    }
+    message Level {
+      // An optional identifier which can be used to reference the component elsewhere in the BOM. Uniqueness is enforced within all elements and children of the root-level bom element.
+      optional string bom_ref = 1;
+      // The identifier used in the standard to identify a specific level.
+      optional string identifier = 2;
+      // The title of the level.
+      optional string title = 3;
+      // The description of the level.
+      optional string description = 4;
+      // The list of requirement `bom-ref`s that comprise the level.
+      repeated string requirements = 5;
+    }
+    // An optional identifier which can be used to reference the component elsewhere in the BOM. Uniqueness is enforced within all elements and children of the root-level bom element.
+    optional string bom_ref = 1;
+    // The name of the standard. This will often be a shortened, single name of the standard.
+    optional string name = 2;
+    // The version of the standard.
+    optional string version = 3;
+    // The description of the standard.
+    optional string description = 4;
+    // The owner of the standard, often the entity responsible for its release.
+    optional string owner = 5;
+    // The list of requirements comprising the standard.
+    repeated Requirement requirements = 6;
+    // The list of levels associated with the standard. Some standards have different levels of compliance.
+    repeated Level levels = 7;
+    // External references provide a way to document systems, sites, and information that may be relevant, but are not included with the BOM. They may also establish specific relationships within or external to the BOM.
+    repeated ExternalReference externalReferences = 8;
+  }
+  repeated Standard standards = 1;
+}
+
+message CryptoProperties {
+  enum CryptoAssetType {
+    CRYPTO_ASSET_TYPE_ALGORITHM = 0;
+    CRYPTO_ASSET_TYPE_CERTIFICATE = 1;
+    CRYPTO_ASSET_TYPE_PROTOCOL = 2;
+    CRYPTO_ASSET_TYPE_RELATED_CRYPTO_MATERIAL = 3;
+  }
+  message AlgorithmProperties {
+    enum CryptoPrimitive {
+      CRYPTO_PRIMITIVE_DRBG = 0;
```

Comment:
this offends proto3 !! 
value 0 is to be reserved for the purpose of "unspecified"/"fallback"/"unmapped".

i will have this fixed via https://github.com/CycloneDX/specification/issues/384

> All enums should have a zero value. proto3 doesn't differentiate between set and unset fields, so if an enum field isn't explicitly set, it defaults to the zero value. If an explicit zero value isn't part of the enum definition, this defaults to the actual zero value of the enum. For example, if you had the following .proto file, any Uri with scheme not explicitly set would default to SCHEME_FTP:

Reply:
about to be fixed via https://github.com/CycloneDX/specification/pull/385